### PR TITLE
Move firebase test lab logs to a staging bucket

### PR DESCRIPTION
**Pull request by ricardoamador**
## Description

Save logs to staging for firebase tasks.

## Issues

This addresses and OKR for this quarter for security improvements.

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by keyonghan** [comment]
@dnfield Are we still running firebase test lab test in engine CI?

**Comment by dnfield** [comment]
We run firebase test lab but we don't use Cirrus at all.. 

This script gets invoked from https://flutter.googlesource.com/recipes/+/refs/heads/main/recipes/engine/engine.py#547

**Comment by keyonghan** [comment]
> We run firebase test lab but we don't use Cirrus at all..
> 
> This script gets invoked from https://flutter.googlesource.com/recipes/+/refs/heads/main/recipes/engine/engine.py#547

Got it. For this case, we will need to update the bucket then.

For context: we are moving the log location from `flutter_firebase_testlab` bucket to a new bucket `flutter_firebase_testlab_staging`. No breakage is expected, but please let us know if you see any issues after.

**Review by keyonghan — APPROVED** [review]
LGTM

**Comment by keyonghan** [comment]
Updated title and body to remove cirrus keywords.

File: ci/firebase_testlab.py
Comment (dnfield) [inline ci/firebase_testlab.py:14]: nit: should this just be an env variable coming from LUCI?

Comment (keyonghan) [reply ci/firebase_testlab.py:14]: SGTM. Let's put the project as an env as well. /cc @ricardoamador

Comment (ricardoamador) [reply ci/firebase_testlab.py:14]: On it.

**Comment by ricardoamador** [comment]
@keyonghan @dnfield to circle back the bucket and project have been updated in the firebase_testlab.py script in the engine repo. Do we need to change the bucket in the invoking recipe script? That one is using flutter_infra_release.

**Comment by keyonghan** [comment]
> @keyonghan @dnfield to circle back the bucket and project have been updated in the firebase_testlab.py script in the engine repo. Do we need to change the bucket in the invoking recipe script? That one is using flutter_infra_release.

After supporting env here, we need to update recipes to setup these two variable values there.

**Comment by keyonghan** [comment]
We should hold this until the recipes env support.

File: ci/firebase_testlab.py
Comment (dnfield) [inline ci/firebase_testlab.py:15]: These should either get defaulted to something sensible or there should be an error thrown if they're not set.

Comment (ricardoamador) [reply ci/firebase_testlab.py]: Done.

**Review by dnfield — APPROVED** [review]

**Comment by ricardoamador** [comment]
The Linux Android AOT Engine test will fail until the recipes change has been merged.
https://flutter-review.googlesource.com/c/recipes/+/32381

**Pull request merged**
